src/components/modebar/buttons.js

```diff
@@ -347,15 +347,19 @@ function handleCamera3d(gd, ev) {
         var key = sceneId + '.camera';
         var scene = fullLayout[sceneId]._scene;
 
-        if(attr === 'resetDefault') {
+        if(attr === 'resetDefault' || attr === 'resetLastSave') {
             aobj[key] = Lib.extendDeep({}, scene.cameraInitial);
```

Comment:
Wouldn't it be easier to _just_ save keys `up`, `eye` and `center` in `scene.cameraInitial` instead of saving the all `scene.camera` keys and then doing this fancy footwork to get the correct `projection.type`?

In other words, do we really need a `cameraInitial` stash? Could we instead just use a "view" stash:

```js
scene.viewInitial = {
  up: {x: /**/, y: /**/, z: /**/},
  eye: {x: /**/, y: /**/, z: /**/},
  center: {x: /**/, y: /**/, z: /**/}
}
```

Reply:
Good call. 
Done in https://github.com/plotly/plotly.js/pull/3597/commits/9e1aa7af86a8dab6e807e8ddea8dca826031cc41.